src/main/java/marquez/api/resources/JobResource.java

```diff
@@ -214,17 +211,15 @@ public Response abortJobRun(@PathParam("runId") final String runId) throws Resou
     }
   }
 
-  @GET
-  @Produces(APPLICATION_JSON)
+  @PUT
   @Timed
-  @Path("/jobs/runs/{runId}")
-  public Response get(@PathParam("runId") final UUID runId) throws ResourceException {
+  @Path("/jobs/runs/{runId}/abort")
+  public Response abortJobRun(@PathParam("runId") final String runId) throws ResourceException {
     try {
-      Optional<marquez.core.models.JobRun> jobRun = jobService.getJobRun(runId);
+      final Optional<JobRun> jobRun = jobService.getJobRun(UUID.fromString(runId));
       if (jobRun.isPresent()) {
-        return Response.status(Response.Status.OK)
-            .entity(new CoreJobRunToApiJobRunMapper().map(jobRun.get()))
-            .build();
+        jobService.updateJobRunState(UUID.fromString(runId), JobRunState.State.ABORTED);
```

Comment:
interesting...this is the resource layer reaching into the `JobRunState` core model. `JobService` might need to make this easier for the resource layer

Reply:
Yea, agreed, it almost feels like breaking encapsulation. What we have is good, but it would feel more natural to ask the service to mark the job as {failed, aborted, completed}, but not prescribing that it should cause this field to change value.